dhall/src/Dhall/Main.hs

```diff
@@ -540,19 +551,51 @@ noHeaders
 noHeaders i =
     i
 
+data Config = Config
+    { ascii  :: Maybe Bool } deriving Dhall.Generic
+
+instance Dhall.FromDhall Config
+
 -- | Run the command specified by the `Options` type
 command :: Options -> IO ()
 command (Options {..}) = do
-    let characterSet = case ascii of
-            True  -> ASCII
-            False -> Unicode
 
     GHC.IO.Encoding.setLocaleEncoding System.IO.utf8
 
     let rootDirectory = \case
             InputFile f   -> System.FilePath.takeDirectory f
             StandardInput -> "."
 
+    let getCharacterSet input = case optionsAscii of
+            Just b  -> return $ toCharSet b
+            Nothing -> do
+                configPath <- findConfig $ rootDirectory input
+                config <- decode configPath
+                let isAscii = fromMaybe False $ configToIsAscii config in
+                    return . toCharSet $ isAscii
+            where
+                toCharSet True = ASCII
+                toCharSet False = Unicode
+
+                configToIsAscii Nothing = Just False
+                configToIsAscii (Just config) = ascii config
+
+                decode (Just s) = fmap Just $ Dhall.input Dhall.auto $ Data.Text.pack s
+                decode Nothing = return Nothing
+
+                findConfig startDir =
+                    System.Directory.makeAbsolute startDir >>= \case
+                            "/" -> return Nothing
```

Comment:
It seems like we should technically allow the configuration file to reside in `/`, just for consistency